### PR TITLE
Creation of event based on an existing one, including activities

### PR DESCRIPTION
**Pull request by thermvs**

### CODE REVIEW

**Comment by IP-13** [comment]
|Overall Project|59.9% **`-0.74%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[EventController.java](https://github.com/itmo-events-app/event-app/blob/48afa43ff8aee9c1347a8b1fd8a5f498c4ceb407/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FEventController.java)|89.47% **`-10.53%`**|:x:|
|[EventService.java](https://github.com/itmo-events-app/event-app/blob/48afa43ff8aee9c1347a8b1fd8a5f498c4ceb407/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventService.java)|82.97% **`-7.28%`**|:x:|

File: src/main/java/org/itmo/eventapp/main/service/EventService.java [hunk 1]
```diff
@@ -255,4 +255,22 @@ public Event copyEventByOne(Event existingEvent, Event parentEvent) {
         minioService.copyImagesWithPrefix(BUCKET_NAME, BUCKET_NAME, imagePrefix, newImagePrefix);
         return savedEvent;
     }
+
+    @Transactional
+    public Event createEventBasedOnExistingWithNewTitleAndAdmin(Integer eventId, String title, Integer userId) {
+        Event existingEvent = findById(eventId);
+        Event newEvent = copyEventByOne(existingEvent, existingEvent.getParent());
+        newEvent.setTitle(title);
```
Comment (anamrzv) [inline src/main/java/org/itmo/eventapp/main/service/EventService.java:263]: не вызывается save после установки нового названия

File: src/main/java/org/itmo/eventapp/main/service/EventService.java [hunk 2]
```diff
@@ -255,4 +255,22 @@ public Event copyEventByOne(Event existingEvent, Event parentEvent) {
         minioService.copyImagesWithPrefix(BUCKET_NAME, BUCKET_NAME, imagePrefix, newImagePrefix);
         return savedEvent;
     }
+
+    @Transactional
+    public Event createEventBasedOnExistingWithNewTitleAndAdmin(Integer eventId, String title, Integer userId) {
+        Event existingEvent = findById(eventId);
+        Event newEvent = copyEventByOne(existingEvent, existingEvent.getParent());
```
Comment (anamrzv) [inline src/main/java/org/itmo/eventapp/main/service/EventService.java:262]: тут надо taskService.copyEventWithTasks(id, deep); 
чтобы скопировались задачи тоже

File: src/main/java/org/itmo/eventapp/main/service/EventService.java (hunk 1 above)
Comment (thermvs) [reply src/main/java/org/itmo/eventapp/main/service/EventService.java:263]: у меня его нет, потому что транзакции можно явно не вызывать `save`, спринг сам сохранит новое название

**Comment by IP-13** [comment]
|Overall Project|60.13% **`-0.6%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[EventController.java](https://github.com/itmo-events-app/event-app/blob/c48166e52963eff17d682aa38530e1d1fcfcfbe9/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FEventController.java)|88.7% **`-11.3%`**|:x:|
|[EventService.java](https://github.com/itmo-events-app/event-app/blob/c48166e52963eff17d682aa38530e1d1fcfcfbe9/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FEventService.java)|86.31% **`-3.54%`**|:x:|
|[TaskService.java](https://github.com/itmo-events-app/event-app/blob/c48166e52963eff17d682aa38530e1d1fcfcfbe9/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FTaskService.java)|75.6% **`-1.5%`**|:x:|

**Review by IP-13 — APPROVED** [review]

**Pull request merged**